src/h5web/visualizations/index.test.ts

```diff
@@ -287,8 +306,253 @@ describe('Visualization definitions', () => {
         { ...baseDatatype, type: '' },
       ];
 
-      unsupportedEntities.forEach((dataset) => {
-        expect(supportsEntity(Vis.Heatmap, dataset)).toBe(false);
+      unsupportedEntities.forEach((entity) => {
+        expect(supportsEntity(Vis.Heatmap, entity)).toBe(false);
+      });
+    });
+  });
+
+  describe('Vis.NxSpectrum', () => {
+    const nxSpectrumDef = VIS_DEFS[Vis.NxSpectrum];
+
+    it('should support NXdata group referencing signal with spectrum interpretation', () => {
+      const metadata = {
+        datasets: {
+          [baseDataset.id]: {
+            ...baseDataset,
+            attributes: [
+              { name: 'interpretation', value: NxInterpretation.Spectrum },
+            ],
+          },
+        },
+      } as HDF5Metadata;
+
+      const supported = nxSpectrumDef.supportsEntity(nxDataGroup, metadata);
+      expect(supported).toBe(true);
+    });
+
+    it('should support NXdata group referencing signal with numeric type, simple shape, one dimension and no or unknown interpreation', () => {
+      const supportedSignals: HDF5Entity[] = [
+        {
+          ...baseDataset,
+          type: { class: HDF5TypeClass.Integer, base: 'H5T_STD_I32LE' },
+          shape: { class: HDF5ShapeClass.Simple, dims: [1] },
+          attributes: [], // no interpretation
+        },
+        {
+          ...baseDataset,
+          type: { class: HDF5TypeClass.Float, base: 'H5T_IEEE_F32BE' },
+          shape: { class: HDF5ShapeClass.Simple, dims: [1] },
+          attributes: [
+            { name: 'interpretation', value: 'unknown' } as HDF5Attribute, // unknown interpretation
+          ],
+        },
```

Comment:
Given how interpretations are handled for now, I would also add a 2D dataset with unknown interpretation to the supported datasets.

Reply:
If the interpretation is unknown, it falls back to the dimension check. Since a 2D dataset with an unknown interpretation is supported by `Vis.NxImage`, then it doesn't need to be supported by `Vis.NxSpectrum`. That's why the check is different than for the line vis: `Vis.NxSpectrum` only supports 1D datasets. We had discussed this with Thomas, if I recall.

Reply:
> Vis.NxSpectrum only supports 1D datasets, not nD datasets.